### DISCUSSION TITLE
Need help with understanding the core advantage of sveltekit-i18n over other libraries

### DISCUSSION THREAD

**Discussion by aakash14goplani** [sveltekit-i18n/lib#128] [Q&A] [answered]
Hello @jarda-svoboda and other community members,

I am migrating my projects from Svelte to SvelteKit and exploring options for i18n. As of today we use svelte-i18n in our Svelte-based projects. While exploring I came across this package, which at initial view, looks like extension of svelte-i18n with few add-ons. Just to confirm whether my understanding is correct and get my facts straight about this library, could you help me with following questions:

1. What was core purpose of creating this library or to rephrase this what problem does this library solves that others (for example svelte-i18n cannot) ?
2. Can you please explain with example how sveltekit-i18n is providing these features and others fail to do so:
   - Module-based: your translations are loaded for visited pages only (and only once!)
   - Component-scoped translations: you can create multiple instances with custom definitions
   - Custom modifiers: you can modify the input data the way you really need

**Comment by jarda-svoboda** [comment, marked as answer]
Hi @aakash14goplani, thanks for your great question!
Yes, this library is heavily inspired by `svelte-i18n`, but it provides more flexible solution in terms of loading translations and the message syntax. Long story short:

1) I've created this library because at that time, there wasn't any other library, which would be able to load my translation definitions dynamically on demand and effectively only at the time i really need concrete scope of translations for concrete locale. 
Also, fetching translation definitions from the language server (or external service) was impossible with most of other libraries.

2) Let me explain you, what these mean:
   - Module-based translations - this means, that every translation definition can be loaded as separate module (or scope) depending on given path. That's good for performance on node-based applications. You don't have to load all translations at once for the whole site, but by pieces at the time it's needed. 
   Try to run [this example](https://github.com/sveltekit-i18n/lib/tree/master/examples/multi-page) locally. There is a console log message saying `Loading translations...` informing you when your translation definitions are loaded. You should get maximum of 4 of these messages. Once on server (during the SSR), and then once for every page you visited (in your browser).
   
   - Component-scoped translations means, that you can have separate translations for components, if needed. This loads your component's translation definitions only in case the component is included within visited page. 
   You can find an example [here](https://component-scoped-csr.netlify.app/).
   
   - The default message parser's abilities are presented [here](https://parser-default.netlify.app/) including custom modifiers (learn more about the default parser [here](https://github.com/sveltekit-i18n/parsers/tree/master/parser-default)). Anyway, you don't have to go with the `sveltekit-i18n`'s default message syntax. ICU syntax is also supported if needed - learn more [here](https://github.com/sveltekit-i18n/parsers/tree/master/parser-icu)

Hope, this helped you to understand how this lib works. Let me, please, know if you have something else to clarify.)
